shared_memory_client.py

```diff
@@ -0,0 +1,170 @@
+        return loads(self._connection.receive(self.BUFFER_SIZE)).response_code
+
+    def oldestRx(self, smt_object, data_info_object=None):
+        if data_info_object is None:
+            request = Request(self.SMT_GET_OLDEST, smt_object)
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            return response.response_code
+        else:
+            request = Request(self.SMT_GET_OLDEST, (smt_object, data_info_object))
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj, temp_data = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            self._copy_shared_memory_object(temp_data, data_info_object)
+            return response.response_code
+
+    def getOldest(self, smt_object, data_info_object=None):
+        if data_info_object is None:
+            request = Request(self.SMT_GET_OLDEST, smt_object)
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp = response.response_object
+            self._copy_shared_memory_object(temp, smt_object)
+            return response.response_code
+        else:
+            request = Request(self.SMT_GET_OLDEST, (smt_object, data_info_object))
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj, temp_data = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            self._copy_shared_memory_object(temp_data, data_info_object)
+            return response.response_code
+
+    def receive(self, smt_object, counter, timeout, data_info_object=None):
+        if data_info_object is None:
+            request = Request(self.SMT_GET_BY_COUNTER, [smt_object, counter, timeout])
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_object = response.response_object
+            self._copy_shared_memory_object(temp_object, smt_object)
+            return response.response_code
+        else:
+            request = Request(self.SMT_GET_BY_COUNTER, [smt_object, counter, timeout, data_info_object])
+            self._connection.send(dumps(request))
+            response = loads(self._connection.receive(self.BUFFER_SIZE))
+            temp_obj, temp_data = response.response_object
+            self._copy_shared_memory_object(temp_obj, smt_object)
+            self._copy_shared_memory_object(temp_data, data_info_object)
+            return response.response_code
+
+    def getByCounter(self, smt_object, counter, timeout, data_info_object=None):
```

Comment:
What is the difference between this and 'receive'?

Reply:
Avisar told me to create two functions with the same implementation, in order to let the testers choose whatever they want to call it.